### ISSUE TITLE
User settings wiped

### ISSUE THREAD

**Issue by motebotic** [Microsoft/vscode#26381]
- VSCode Version: Code 1.12.1 (f6868fce3eeb16663840eb82123369dec6077a9b, 2017-05-04T21:40:39.245Z)
- OS Version: Linux x64 4.4.0-77-generic
- Extensions: the listing exceeds the lower minimum of browsers' URL characters limit
---

I cant reproduce the bug so this is probably unhelpful, but here's how it happened:

1. Pressed ctrl + to zoom & got a message VS needs to restart to apply settings. 
2. Clicked cancel (was on a roll).
3. VS reset to default everything on clicking cancel.
4. Closed VS & reopened but the settings where still the default.
5. Checked .config settings json and it was certainly wiped.

Cheers :)

**Comment by motebotic** [comment]
I have made a new settings file and tried to reproduce this but can't even get the "need to restart to apply settings" when I am zooming. I tried hitting all the keys that my fat fingers may have slipped to and nothing. The file I was trying to zoom on is a file I keep a bunch of text symbols in to use when boiler-plating shit if that changes anything. 

Anyway I am sure there are better things to think & spend time on than this. I will let you know if this happens again.

Thanks for awesome editor btw!

**Comment by bpasero** [comment]
I think @jrieken saw this too, at least the dialog (unsure if settings got lost)?

**Comment by jrieken** [comment]
Correct, I and @alexandrudima have seen this when zooming in to take a closer look at the cursor. I do have the native window style configure to be native (`"window.titleBarStyle": "native"`) but I don't think that hand an impact. I was also afraid about having lost my changed because until restart the default theme was used. 

![screen shot 2017-05-09 at 11 17 01](https://cloud.githubusercontent.com/assets/1794099/25903201/098d514e-359c-11e7-8b66-55d0ce9cdc75.png)

**Comment by alexandrudima** [comment]
I saw it on @jrieken 's machine. All he did was to do `ctrl+=` (i.e. zoom in) many times to zoom in multiple times.

**Comment by sandy081** [comment]
Duplicates #25427

**Comment by sandy081** [comment]
@motebotic Very sad to know that settings are wiped out. I already have an issue #25427 opened for this and collecting data about this. Pain point is that, it was not reproducible and I am try to figure out what could cause this. Thanks for reporting it.

**Comment by motebotic** [comment]
@bpasero I think I had "window.titleBarStyle": "native" as I was getting some strange flickering that was driving me nuts some time back & read about that as a cure. Since I don't have my old setting though I cant say for certain. I had turned telemetry.enableCrashReporter off.. sorry that's on now ;) although it didn't crash so it wouldn't have helped.

**Comment by bpasero** [comment]
@sandy081 it looks to me that what happened to @jrieken is that briefly all settings got lost and thus the dialog got triggered (this would also explain why @jrieken saw the default theme being used). So this seems to be the cause, but still does not explain why settings go empty randomly...

**Comment by motebotic** [comment]
@bpasero @sandy081 what could cause vsc to think that a setting had changed in both mine & @jrieken case. I had one window open, split in two- with two JS files on the left side & one on the right side. I was working in file on right & had an invalid JS file holding some glyph's on the left in a hidden tab. I clicked the valid JS file on the left side, hit ctrl tab to change to the icon file & started zooming in when the dialog popped up. The zoom was slow to respond when I hit it the first time & being inpatient & on a roll I hit it multiple times again before the dialog happened. When I cancelled the dialog & seen the default theme appear, side bar etc I hit ctrl, to check the settings.json & seen it was empty all but the window zoom level. I closed the settings files & restarted VS code hoping my settings would come back, they did not. I tried to reboot & still empty, googled, checked the settings.json in .config/.... & that file only contained the window zoom level.

Risking coming off stupid here.. C=could it be something to do with trying to write to the settings.json as something else is going on & returns that the file doesn't exist? When I changed focus from the right to left my file would have automatically saved & I was moving quick. I have no idea how this works just throwing things out there. 

So yeah hope that was of some help, or at least made .. sense ((:
